src/main/java/core/characters/CharacterManager.java

```diff
@@ -1,80 +1,145 @@
 package core.characters;
 
+import java.lang.reflect.Array;
 import java.util.ArrayList;
-
-// Need to import InputManager and ControlState
+import java.util.HashMap;
+import java.util.Objects;
 
 public class CharacterManager {
 
     /*
      * Use case class that handles updating instances of GameCharacter based on inputs from the InputHandler
      * @param character: An instance of GameCharacter being updated
+     * @param type: Either Player or AI so the input handler can distinguish between usages
      * TODO: Change item to type Item when implemented
-     * TODO: Import InputManager and ControlState
      * */
 
-    private GameCharacter character;
+    public HashMap<Integer, GameCharacter> characterEntities;
+    private ArrayList<Integer> keylist;
+
+    public CharacterManager() {
+        this.characterEntities = new HashMap<Integer, GameCharacter>();
+        this.keylist = new ArrayList<Integer>();
+    }
 
-    public CharacterManager(GameCharacter character) {
-        this.character = character;
+    public void addCharacter(GameCharacter character) {
+        /*
+        * Player will always be hashed as 0
+        * Defenders will be hashed as even ints
+        * Enemies will be hashed as odd ints
```

Comment:
player as 0 seems strict – what if we ever want 2 players, for example? very change resistant imo

and what if we want a completely different class of npc, etc etc

imo it's more simple to have a unique ID for everything whose only purpose is that – being a unique ID. otherwise this leads to difficulty altering anything in the future since there's lots of side effects